### ISSUE TITLE
Failed to build  "bash apollo.sh build

### ISSUE THREAD

**Issue by vinsenvan** [ApolloAuto/apollo#266]
![image](https://user-images.githubusercontent.com/30096640/28650268-2e007742-72ad-11e7-9dc9-aaefcc94cbd3.png)

platform: Ubuntu 14.04 LTS

command: bash apollo.sh build

**Comment by vinsenvan** [comment]
I have try to run  "bash apollo.sh clean" before,but it cann‘t solve this problem

**Comment by vinsenvan** [comment]
![image](https://user-images.githubusercontent.com/30096640/28656037-2c82e8da-72d2-11e7-89b3-fc2c42ade5d4.png)
now，the apollo.sh build is ok，but I can’t open the page with localhost：8887，what should i do？

**Comment by siyangy** [comment]
Build doesn't seem to have succeeded, notice that there was an error above and bazel found 0 target to build. This issue actually has already been fixed, can you pull the latest code and try again?

**Comment by vinsenvan** [comment]
![default](https://user-images.githubusercontent.com/30096640/28760783-25a715ba-75dc-11e7-9b01-0813c386fde5.png)
it seem to build succeeded after pull the latest code ,but after i click into Dreamview , it set up 8888 ok, but there wasn't a car in the screem,what should I do?
and I wond if it is needed to build "bash apollo.sh build" every times before  I run "bash scripts/hmi.sh"?

**Comment by siyangy** [comment]
Please refer to https://github.com/ApolloAuto/apollo/blob/master/README.md for a step by step guide. You can run a demo by running rosbag play -l ./docs/demo_guide/demo.bag.

It's not necessary to build unless you have changed some code and want to re-compile.

**Comment by vinsenvan** [comment]
![image](https://user-images.githubusercontent.com/30096640/28761385-b3eaf7a2-75e0-11e7-84b7-e93256b8b804.png)
it is ok after  running rosbag play -l ./docs/demo_guide/demo.bag, see below:
![image](https://user-images.githubusercontent.com/30096640/28761481-72752d6e-75e1-11e7-8e34-d378a41182c7.png)
thank you  very much!

**Comment by WintonHuang** [comment]
Hello, I run "bash apollo.sh clean", then run  "bash apollo.sh build". but still have the same issue. I want to know how to solve it???  thanks

**Comment by WintonHuang** [comment]
INFO: Found 189 targets...
ERROR: /home/winton/.cache/bazel/_bazel_winton/540135163923dd7d5820f3ee4b306b32/external/boringssl/boringssl-chromium-stable-with-bazel/BUILD:92:1: undeclared inclusion(s) in rule '@boringssl//boringssl-chromium-stable-with-bazel:crypto':
this rule is missing dependency declarations for the following files included by 'external/boringssl/boringssl-chromium-stable-with-bazel/src/crypto/fipsmodule/bcm.c':
  '/home/winton/.cache/bazel/_bazel_winton/540135163923dd7d5820f3ee4b306b32/external/boringssl/boringssl-chromium-stable-with-bazel/src/crypto/fipsmodule/aes/aes.c'
  '/home/winton/.cache/bazel/_bazel_winton/540135163923dd7d5820f3ee4b306b32/external/boringssl/boringssl-chromium-stable-with-bazel/src/crypto/fipsmodule/aes/key_wrap.c'

**Comment by huangwen0907** [comment]
@siyangy HI, I want to ask a question:
If I followed the guide [here](https://github.com/ApolloAuto/apollo/blob/master/README.md) and it can build/set the apollo successfully at the first time.
 But if I update the code in the apollo folder by the command below:
>git fetch origion
git check origin/master
git checkout -B master  

 then I can get the newest update code from the github.  But at this time all the files which i download at the time i first set the environment is disappear (Btw: I set the environment in the apollo folder). 
I know I make some mistakes, but i do not know how to do: if I am not the first time start the appolo simulation, how can i quick start it and how to solve the disappear files problems. Thank you very much.

**Comment by siyangy** [comment]
It's quite hard to follow what you mean here. Can you be more specific about what files are missing and what you mean by 'I set the environment in the apollo folder'? I am also not sure why you ran the three git commands listed. You will get the latest code at master branch after the initial git clone and after that you should do a git pull and build the code again.
